apexlog/apexlog.go

```diff
@@ -0,0 +1,143 @@
+package apexlog
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/airbrake/gobrake/v5"
+
+	"github.com/apex/log"
+)
+
+// severity identifies the sort of log: info, warning etc. It also implements
+// the flag.Value interface. The -stderrthreshold flag is of type severity and
+// should be modified only through the flag.Value interface. The values match
+// the corresponding constants in C++.
+type severity int8
+
+// These constants identify the log levels in order of increasing severity.
+// A message written to a high-severity log file is also written to each
+// lower-severity log file.
+const (
+	debugLog severity = iota
+	infoLog
+	warningLog
+	errorLog
+	fatalLog
+)
+
+var severityName = []string{
+	debugLog:   "debug",
+	infoLog:    "info",
+	warningLog: "warning",
+	errorLog:   "error",
+	fatalLog:   "fatal",
+}
+
+func severityByName(s string) (severity, bool) {
+	for i, name := range severityName {
+		if name == strings.ToLower(s) {
+			return severity(i), true
+		}
+	}
+	return 0, false
+}
+
+var gobrakeSeverity = errorLog
+
+// SetGobrakeSeverity sets minimum log severity that will be sent to Airbrake.
+//
+// Valid names are "debug" "info", "warning", "error", and "fatal".  If the name is not
+// recognized, "error" severity is used.
+func setGobrakeSeverity(severity string) {
+	name, ok := severityByName(severity)
+	if ok {
+		gobrakeSeverity = name
+	}
+}
+
+// Gobrake is an instance of Airbrake Go Notifier that is used to send
+// logs to Airbrake.
+var Gobrake *gobrake.Notifier
+
+// Handler implementation.
+type Handler struct {
+	Gobrake *gobrake.Notifier
+}
+
+// New Apex Logs handler for airbrake.
+//
+// Valid severity levels are "debug" "info", "warning", "error", and "fatal".  If the name is not
+// recognized, "error" severity is used.
+func New(notifier *gobrake.Notifier, severity string) *Handler {
+	setGobrakeSeverity(severity)
+	var v Handler
+	Gobrake = notifier
```

Comment:
What will happen if two projects use this `apexlog` integration? Probably only one integration will win and the other one will stop sending logs. Can we make sure we support this scenario?

I am actually not sure why we store `notifier` on package-level. We store it on the handler itself on the next line, so why not to use it from there?